### PR TITLE
Draft: 115/provider connects with indexer

### PR DESCRIPTION
**Pull request by alvaro-alonso**
As I've seen that the provider node is not connected with the indexers DB, I've created this branch.  Is someone working on the connection? If no one is doing, I can take that over or lmk if I can assist in this

This branch haven't not yet been tested. I just got the FAUCET credentials. I will do that.

- [ ] I would also like to rename the directory back from `backend` to `provider`

### CODE REVIEW

File: backend/app/main.py [hunk 1]
```diff
@@ -33,54 +36,15 @@ class AuthData(BaseModel):
...
-    pbk = scrub_input(data.pbkey)
-    pbk2 = base58_decode(pbk)
-    pkh = blake2b(data=pbk2, digest_size=20).digest()
-    prefix = b'tz1'
-    pbh2 = base58_encode(pkh, prefix).decode()
-    pbh_of_buyer = ""  # to be called from database
-    print("Check matching onchain PBK from buyer with consumer pbk..")
-    if pbh2 == pbh_of_buyer:
-        print("Verified account!")
+    if signature_verification(**data, db):
```
Comment (amecklenborg) [inline backend/app/main.py:43]: positional argument not working

File: backend/app/main.py [hunk 2]
```diff
@@ -33,54 +36,15 @@ class AuthData(BaseModel):
...
-
 # get digital twin data by unique id
 @app.post("/download/{unique_id}")
-async def fetch_data(unique_id: int, data: AuthData):
-    results = {"unique_id": unique_id}
-    # results = {"unique_id": unique_id, "nft_id": data.nft_id, "sig": data.sig, "public key": data.pbkey}
-    results.update({"data": data})
-    print(results)
-
+async def fetch_data(unique_id: int, data: AuthData, db: Session = Depends(get_db)):
```
Comment (amecklenborg) [inline backend/app/main.py:41]: get_db? Unresolved reference?

File: backend/app/utils.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    """Creates base58 encoded public key hash for this key.
+
+    :returns: the public key hash for this key
+    """
+    pkh = blake2b(data=public_point, digest_size=20).digest()
+    prefix = b'tz1'
+    return base58_encode(pkh, prefix).decode()
+
+
+def signature_verification(nft_id: str, sig: str, pbkey: str, db: Session) -> None:
```
Comment (amecklenborg) [inline backend/app/utils.py:23]: return Bool

**Comment by alvaro-alonso** [comment]
I close these branch in favor #143

**Pull request closed without merging**